src/main/java/org/jboss/ejb/client/EJBRootContext.java

```diff
@@ -83,9 +83,15 @@ protected Object lookupNative(final Name name) throws NamingException {
             }
         }
         if (beanName == null) {
+            if (size == 3) {
+                // name is of the form appName/moduleName/distinctName
+                return new RelativeContext(new FastHashtable<>(getEnvironment()), this, SimpleName.of(name));
+            }
```

Comment:
I think that this suffers from a problem where you cannot disambiguate between app/module/distinct vs app/module/bean, doesn't it?